### PR TITLE
Outlets - Added tests for uncontrolled outlets

### PR DESCRIPTION
**Pull request by zack-rma**
Adds in test cases for uncontrolled outlets

### CODE REVIEW

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+import static cwms.cda.data.dao.DaoTest.getDslContext;
+import static org.junit.jupiter.api.Assertions.*;
+
+@Tag("integration")
+class OutletDaoUncontrolledIT extends ProjectStructureIT {
```
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Rename to OutletDaoRatingIT

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                    DeleteRule.DELETE_ALL);
+            outletDao.deleteOutlet(SPILLWAY_LOW_FLOW_2_LOC.getOfficeId(), SPILLWAY_LOW_FLOW_2_LOC.getName(),
+                    DeleteRule.DELETE_ALL);
+            deleteLocation(context, SPILLWAY_LOW_FLOW_1_LOC.getOfficeId(), SPILLWAY_LOW_FLOW_1_LOC.getName());
+            deleteLocation(context, SPILLWAY_LOW_FLOW_2_LOC.getOfficeId(), SPILLWAY_LOW_FLOW_2_LOC.getName());
+        }, CwmsDataApiSetupCallback.getWebUser());
+        tearDownProject();
+    }
+
+    @Test
```
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: To 198, we don't need, this is already tested

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+import java.io.IOException;
+import java.util.List;
+
+import static cwms.cda.data.dao.DaoTest.getDslContext;
+import static org.junit.jupiter.api.Assertions.*;
+
+@Tag("integration")
+class OutletDaoUncontrolledIT extends ProjectStructureIT {
+    private static final String OUTLET_KIND = "OUTLET";
+    private static final CwmsId SPILLWAY_LOW_FLOW_RATING_GROUP = new CwmsId.Builder()
```
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Rename SPILLWAY to PROJECT for low flow pieces.

Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Make two groups, one for Rating-{project}-Spillway, and one for Rating-{project}-LowFlow

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    static void setup() throws Exception {
+        setupProject();
+        CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
+        databaseLink.connection(c -> {
+            DSLContext context = getDslContext(c, OFFICE_ID);
+            OutletDao outletDao = new OutletDao(context);
+            try {
+                storeLocation(context, SPILLWAY_LOW_FLOW_1_LOC);
+                storeLocation(context, SPILLWAY_LOW_FLOW_2_LOC);
+                outletDao.storeOutlet(SPILLWAY_LOW_FLOW_1_OUTLET, false);
```
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Don't store these in setup

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        }, CwmsDataApiSetupCallback.getWebUser());
+    }
+
+    @AfterAll
+    static void cleanup() throws Exception {
+        CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
+        databaseLink.connection(c -> {
+            DSLContext context = getDslContext(c, OFFICE_ID);
+            OutletDao outletDao = new OutletDao(context);
+            outletDao.deleteOutlet(SPILLWAY_LOW_FLOW_1_LOC.getOfficeId(), SPILLWAY_LOW_FLOW_1_LOC.getName(),
```
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Don't delete on cleanup

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    private static final CwmsId SPILLWAY_LOW_FLOW_RATING_GROUP = new CwmsId.Builder()
+            .withName("Rating-" + PROJECT_1_ID.getName() + "-LowFlow").withOfficeId(PROJECT_1_ID.getOfficeId()).build();
+    private static final CwmsId SPILLWAY_LOW_FLOW_RATING_GROUP_MODIFIED = new CwmsId.Builder()
+            .withName("Rating-" + PROJECT_1_ID.getName() + "-LowFlow-Modified")
+            .withOfficeId(PROJECT_1_ID.getOfficeId()).build();
+    private static final Location SPILLWAY_LOW_FLOW_1_LOC
+            = buildProjectStructureLocation(PROJECT_1_ID.getName() + "-LF1", OUTLET_KIND);
+    private static final Location SPILLWAY_LOW_FLOW_2_LOC
+            = buildProjectStructureLocation(PROJECT_1_ID.getName() + "-LF2", OUTLET_KIND);
+    private static final Location SPILLWAY_LOW_FLOW_3_LOC
```
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Only have two locations

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 2 above)
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Two test cases:  test_uncontrolled_outlet and test_controlled_outlet

- Store outlet (controlled or uncontrolled)
- Verify location group is created
- Modify location group's shared location alias id to be the associated rating curve id.
- Retrieve outlet and verify it's pointing to the same location group
- Delete outlet
- Delete location group

Prerequisites for each, location exists

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 3 above)
Comment (RyanM-RMA) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Need two Strings for rating curve ID's.  one for spillway (elev;flow curve) and one for low flow (opening-low flow,elev;flow).

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 1 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Renamed as suggested

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 2 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Removed and replaced with proper test cases

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 3 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Renamed as suggested

Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Created groups as suggested

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 4 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Removed storage calls from setup

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 5 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Removed deletion on teardown

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 6 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Removed extra locations

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 2 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Created successful test cases following this pattern

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java (hunk 3 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoUncontrolledIT.java]: Created strings, used in modified group creation

**Review by RyanM-RMA — APPROVED** [review]

File: cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoRatingIT.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    private static final String OUTLET_KIND = "OUTLET";
+    private static final CwmsId PROJECT_LOW_FLOW_RATING_GROUP = new CwmsId.Builder()
+            .withName("Rating-" + PROJECT_1_ID.getName() + "-LowFlow").withOfficeId(PROJECT_1_ID.getOfficeId()).build();
+    private static final CwmsId PROJECT_SPILLWAY_RATING_GROUP = new CwmsId.Builder()
+            .withName("Rating-" + PROJECT_1_ID.getName() + "-Spillway")
+            .withOfficeId(PROJECT_1_ID.getOfficeId()).build();
+    private static final Location PROJECT_LOW_FLOW_LOC
+            = buildProjectStructureLocation(PROJECT_1_ID.getName() + "-LF1", OUTLET_KIND);
+    private static final Location PROJECT_SPILLWAY_LOC
+            = buildProjectStructureLocation(PROJECT_1_ID.getName() + "-LF2", OUTLET_KIND);
```
Comment (rma-psmorris) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/location/kind/OutletDaoRatingIT.java:62]: This location's name should not be low flow (LF) and instead should be spillway.

**Review by rma-psmorris — COMMENTED** [review]
Minor revision request

**Pull request merged**